src/actions/loadData.js

```diff
@@ -88,6 +97,9 @@ const fetchDataAndDispatch = (dispatch, url, query, narrativeBlocks) => {
   // fetchJSON(`${charonAPIAddress}request=mainJSON&url=${url}${fetchExtras}`)
   getDataset(`${url}${fetchExtras}`)
     .then((response) => {
+      url = response.url.split('prefix=')[1];
+      window.history.replaceState({}, "", url);
+
```

Comment:
> Note that it doesn't appear that changing the browser window path has to go through Redux at all. 

Correct -- one way to think of them is that they are independent sources of "state". It's on us to keep these two bits of state in sync, and there is always the potential for bugs! This is why all of the URL manipulation is currently happening in one place - the `changeURL.js` middleware. 

Because of this, this should be moved into `changeURL.js` -- this can be achieved by adding the `url` to the dispatch on line 105 and picking it up in the middleware. Note that this is very similar to how the `_url` magic key currently works, except we're no longer getting it from the JSON anymore 🎉

Reply:
I know there's some confusion about `changeURL.js` so I made a [small example branch](https://github.com/nextstrain/auspice/tree/tmp-redirect-example) with a single commit 872872337880c1de8ec6690a824272a52dcaf2df intended to show what I am talking about.

It works with your nextstrain.org server changes (https://github.com/nextstrain/nextstrain.org/pull/25), and uses your approach to separating out the response and conversion to JSON, but performs the URL modification within `changeURL.js`.